GitCommands/RevisionGraph.cs

```diff
@@ -123,83 +123,71 @@ private void ProccessGitLog(CancellationToken taskState)
                 Updated(this, new RevisionGraphUpdatedEventArgs(null));
             _refs = GetRefs().ToDictionaryOfList(head => head.Guid);
 
-            string formatString =
+            const string shaOnlyFormat =
                 /* <COMMIT>       */ CommitBegin + "%n" +
                 /* Hash           */ "%H%n" +
                 /* Parents        */ "%P%n";
-            if (!ShaOnly)
-            {
-                formatString +=
-                    /* Tree                    */ "%T%n" +
-                    /* Author Name             */ "%aN%n" +
-                    /* Author Email            */ "%aE%n" +
-                    /* Author Date             */ "%at%n" +
-                    /* Committer Name          */ "%cN%n" +
-                    /* Committer Email         */ "%cE%n" +
-                    /* Committer Date          */ "%ct%n" +
-                    /* Commit message encoding */ "%e%x00" + //there is a bug: git does not recode commit message when format is given
-                    /* Commit Subject          */ "%s%x00" +
-                    /* Commit Body             */ "%B%x00";
-            }
+
+            const string fullFormat =
+                shaOnlyFormat +
+                /* Tree                    */ "%T%n" +
+                /* Author Name             */ "%aN%n" +
+                /* Author Email            */ "%aE%n" +
+                /* Author Date             */ "%at%n" +
+                /* Committer Name          */ "%cN%n" +
+                /* Committer Email         */ "%cE%n" +
+                /* Committer Date          */ "%ct%n" +
+                /* Commit message encoding */ "%e%x00" + //there is a bug: git does not recode commit message when format is given
+                /* Commit Subject          */ "%s%x00" +
+                /* Commit Body             */ "%B%x00";
 
             // NOTE:
             // when called from FileHistory and FollowRenamesInFileHistory is enabled the "--name-only" argument is set.
             // the filename is the next line after the commit-format defined above.
 
-            string logParam;
-            if (AppSettings.OrderRevisionByDate)
-            {
-                logParam = " --date-order";
-            }
-            else
-            {
-                logParam = " --topo-order";
-            }
+            string branchFilter = BranchFilter;
+            if (!string.IsNullOrWhiteSpace(BranchFilter) && BranchFilter.IndexOfAny(ShellGlobCharacters) != -1)
```

Comment:
--branches is still incorrectly handled. Move this to where --branches is handled below

Reply:
Thanks. Please check whether 0842600 covers what you were seeing.